### ISSUE TITLE
Test for document regression broken: TypeError in featurize_samples()

### ISSUE THREAD

**Issue by tholor** [deepset-ai/FARM#71]
The test for the recently added RegressionProcessor is broken. 
@busyxin Can you please have a look at what caused this? 

```
../farm/data_handler/data_silo.py:39: in __init__
    self._load_data()
../farm/data_handler/data_silo.py:47: in _load_data
    self.data["train"], self.tensor_names = self.processor.dataset_from_file(train_file)
../farm/data_handler/processor.py:369: in dataset_from_file
    self._featurize_samples()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <farm.data_handler.processor.RegressionProcessor object at 0x7f2a0406e320>

    def _featurize_samples(self):
        chunks_to_process = int(len(self.baskets) / self.multiprocessing_chunk_size)
        num_cpus = min(mp.cpu_count(), self.max_processes, chunks_to_process) or 1
        logger.info(
            f"Got ya {num_cpus} parallel workers to featurize samples in baskets (chunksize = {self.multiprocessing_chunk_size}) ..."
        )
    
        try:
            if "train" in self.baskets[0].id:
                train_labels = []
                for basket in self.baskets:
                    for sample in basket.samples:
                        train_labels.append(sample.clear_text["label"])
                scaler = StandardScaler()
                scaler.fit(np.reshape(train_labels, (-1, 1)))
                self.label_list = [scaler.mean_.item(), scaler.scale_.item()]
                # Create label_maps because featurize is called after Processor instantiation
                self.label_maps = [{0:scaler.mean_.item(), 1:scaler.scale_.item()}]
    
        except Exception as e:
            logger.warning(f"Baskets not found: {e}")
    
        with mp.Pool(processes=num_cpus) as p:
            all_features_gen = p.imap(
                self._multiproc_featurize,
                self.baskets,
                chunksize=self.multiprocessing_chunk_size,
            )
    
            for basket_features, basket in tqdm(
                zip(all_features_gen, self.baskets), total=len(self.baskets)
            ):
                for f, s in zip(basket_features, basket.samples):
                    # Samples don't have labels during Inference mode
                    if "label" in s.clear_text:
                        label = s.clear_text["label"]
>                       scaled_label = (label - self.label_list[0]) / self.label_list[1]
E                       TypeError: unsupported operand type(s) for -: 'str' and 'float'

../farm/data_handler/processor.py:908: TypeError
```

**Comment by busyxin** [comment]
The bug should be solved by #74 
Somehow the tests were passing when farm was still at 0.1.2, I should have tested again after rebasing.

**Comment by tholor** [comment]
Merged & Fixed by #74
